src/metpy/io/gempak.py

```diff
@@ -0,0 +1,2830 @@
+                        merged['DRCT'][0] = parts['PPBB']['DRCT'][0]
+                        merged['SPED'][0] = parts['PPBB']['SPED'][0]
+
+        # Merge MAN temperature
+        bgl = 0
+        qcman = []
+        if num_man_levels >= 2 or num_above_man_levels >= 1:
+            if merged['PRES'][0] == self.prod_desc.missing_float:
+                plast = 2000
+            else:
+                plast = merged['PRES'][0]
+
+        if num_man_levels >= 2:
+            for i in range(1, num_man_levels):
+                if (parts['TTAA']['PRES'][i] < plast
+                   and parts['TTAA']['PRES'][i] != self.prod_desc.missing_float
+                   and parts['TTAA']['TEMP'][i] != self.prod_desc.missing_float
+                   and parts['TTAA']['HGHT'][i] != self.prod_desc.missing_float):
+                    for pname, pval in parts['TTAA'].items():
+                        merged[pname].append(pval[i])
+                    plast = merged['PRES'][-1]
+                else:
+                    if parts['TTAA']['PRES'][i] > merged['PRES'][0]:
+                        bgl += 1
+                    else:
+                        # GEMPAK ignores MAN data with missing TEMP/HGHT and does not
+                        # interpolate for them.
+                        if parts['TTAA']['PRES'][i] != self.prod_desc.missing_float:
+                            qcman.append(parts['TTAA']['PRES'][i])
+
+        if num_above_man_levels >= 1:
+            for i in range(num_above_man_levels):
+                if (parts['TTCC']['PRES'][i] < plast
+                   and parts['TTCC']['PRES'][i] != self.prod_desc.missing_float
+                   and parts['TTCC']['TEMP'][i] != self.prod_desc.missing_float
+                   and parts['TTCC']['HGHT'][i] != self.prod_desc.missing_float):
+                    for pname, pval in parts['TTCC'].items():
+                        merged[pname].append(pval[i])
+                    plast = merged['PRES'][-1]
+
+        # Merge MAN wind
+        if num_man_wind_levels >= 1 and num_man_levels >= 1 and len(merged['PRES']) >= 2:
+            for iwind, pres in enumerate(parts['PPAA']['PRES']):
+                if pres in merged['PRES'][1:]:
+                    loc = merged['PRES'].index(pres)
+                    if merged['DRCT'][loc] == self.prod_desc.missing_float:
+                        merged['DRCT'][loc] = parts['PPAA']['DRCT'][iwind]
+                        merged['SPED'][loc] = parts['PPAA']['SPED'][iwind]
+                else:
+                    if pres not in qcman:
+                        size = len(merged['PRES'])
+                        loc = size - bisect.bisect_left(merged['PRES'][1:][::-1], pres)
+                        if loc >= size + 1:
+                            loc = -1
+                        merged['PRES'].insert(loc, pres)
+                        merged['TEMP'].insert(loc, self.prod_desc.missing_float)
```

Comment:
The next iteration will have a more succinct approach to the merging. The first order of business was to get the output to match GEMPAK. With that done, I did find places to eliminate some redundancy. I kept some of it as is simply because it did not look like I would gain anything by using a class method with changing parameters.

Reply:
👍